Tools/TestWebKitAPI/Tests/WebCore/AudioSampleFormat.cpp

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright (C) 2024 Apple Inc. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. AND ITS CONTRIBUTORS ``AS IS''
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+ * THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL APPLE INC. OR ITS CONTRIBUTORS
+ * BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+ * THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "config.h"
+#include "Test.h"
+
+#include <WebCore/AudioSampleFormat.h>
+#include <cstdint>
+#include <wtf/Vector.h>
+
+namespace TestWebKitAPI {
+
+using namespace WebCore;
+using namespace WTF;
+
+#define AUDIO_SAMPLE_WITH_MORE_TESTS 1
+
+TEST(AudioSampleFormat, U8)
+{
+    auto testVectorUint8 = Vector<uint8_t>::from(0, INT8_MAX / 2, 128, INT8_MAX / 2 + 128, UINT8_MAX);
+    Vector<float> testVectorFloatFromUint8(testVectorUint8.size(), [&](auto index) {
+        return convertAudioElement<float>(testVectorUint8[index]);
+    });
+    Vector<uint8_t> testVectorUint8FromFloat(testVectorFloatFromUint8.size(), [&](auto index) {
+        return convertAudioElement<uint8_t>(testVectorFloatFromUint8[index]);
+    });
+    for (size_t i = 0; i < testVectorUint8.size(); i++)
+        EXPECT_EQ(testVectorUint8[i], testVectorUint8FromFloat[i]);
+
+    // Test that the min and max of the [-1, 1] range can be reached.
```

Comment:
Very optional nit: Comment doesn't quite fit the next 3 lines. Maybe mention 0.0<->128, I guess it's an important value. Or just remove the comment?
And why no comment for the first sub-test above?

& same for the next tests.